File: frontend/lib/types/critique.ts
export interface CritiqueScore {
  score: number;
  max: number;
  note?: string;
}

export interface TopFix {
  priority: string; // "high" | "medium" | "low"
  fix: string;
}

export type CritiqueVerdict = "STRONG MATCH" | "NEEDS WORK" | "WEAK MATCH";
export type CritiqueColor = "green" | "yellow" | "red";

export interface Critique {
  total: number;
  verdict: CritiqueVerdict;
  scores: Record<string, CritiqueScore>;
  keywords_found: string[];
  keywords_missing: string[];
  top_fixes: TopFix[];
}

File: backend/app/models/critique.py
from pydantic import BaseModel, Field


class CritiqueScore(BaseModel):
    score: int
    max: int


class TopFix(BaseModel):
    priority: str  # "HIGH" | "MEDIUM" | "LOW"
    fix: str


class Critique(BaseModel):
    version: int = 1
    scores: dict[str, CritiqueScore] = Field(default_factory=dict)
    total: int
    verdict: str  # "STRONG MATCH" | "NEEDS WORK" | "WEAK MATCH"
    keywords_found: list[str] = Field(default_factory=list)
    keywords_missing: list[str] = Field(default_factory=list)
    top_fixes: list[TopFix] = Field(default_factory=list)

    @staticmethod
    def color_for_score(score: int) -> str:
        if score >= 80:
            return "green"
        if score >= 60:
            return "yellow"
        return "red"

    @staticmethod
    def verdict_for_score(score: int) -> str:
        if score >= 80:
            return "STRONG MATCH"
        if score >= 60:
            return "NEEDS WORK"
        return "WEAK MATCH"
